addons/google_account/google_account.py

```diff
@@ -34,30 +34,32 @@ def generate_refresh_token(self, cr, uid, service, authorization_code, context=N
             content = urllib2.urlopen(req).read()
         except urllib2.HTTPError:
             error_msg = "Something went wrong during your token generation. Maybe your Authorization Code is invalid or already expired"
-            raise self.pool.get('res.config.settings').get_config_warning(cr, _(error_msg), context=context)
+            raise self.env['res.config.settings'].get_config_warning(_(error_msg))
 
         content = simplejson.loads(content)
         return content.get('refresh_token')
 
-    def _get_google_token_uri(self, cr, uid, service, scope, context=None):
-        ir_config = self.pool['ir.config_parameter']
+    @api.model
+    def _get_google_token_uri(self, service, scope):
+        ir_config = self.env['ir.config_parameter']
         params = {
             'scope': scope,
-            'redirect_uri': ir_config.get_param(cr, SUPERUSER_ID, 'google_redirect_uri'),
-            'client_id': ir_config.get_param(cr, SUPERUSER_ID, 'google_%s_client_id' % service),
+            'redirect_uri': ir_config.sudo().get_param('google_redirect_uri'),
+            'client_id': ir_config.sudo().get_param('google_%s_client_id' % service),
             'response_type': 'code',
-            'client_id': ir_config.get_param(cr, SUPERUSER_ID, 'google_%s_client_id' % service),
+            'client_id': ir_config.sudo().get_param('google_%s_client_id' % service),
```

Comment:
See above wrt `ir_config`, it's always sudoed so just set it sudoed in the first place.